daemon/cluster/executor/container/controller.go

```diff
@@ -564,15 +568,9 @@ func (r *controller) matchevent(event events.Message) bool {
 	if event.Type != events.ContainerEventType {
 		return false
 	}
-
-	// TODO(stevvooe): Filter based on ID matching, in addition to name.
-
-	// Make sure the events are for this container.
-	if event.Actor.Attributes["name"] != r.adapter.container.name() {
-		return false
-	}
-
-	return true
+	r.idset.Wait()
```

Comment:
What if `Prepare` is never called? This can happen when an engine is restarted -- the controller will skip the prepare stage.